lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,60 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, options={}) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, options);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, options={}) {
+  if (!options.hasOwnProperty('scope')) {
+    return Node.Status.noChange(node);
+  }
+
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
+    symbolName = node.name;
+  }
+  else if (Node.Type.isUnaryMinus(node)) {
+    symbolName = node.args[0].name;
+  }
+
+  const scope = options.scope;
+  if (scope.hasOwnProperty(symbolName)) {
+    // when declared at top, kept getting
+    // TypeError: <nameIGaveToFunction> is not a function
+    const simplifyExpression = require('../../simplifyExpression');
```

Comment:
so `simplifyExpression` is a pretty high level function - it's the big thing at the top that calls everything else. Because of that I feel weird about using it here.

What you have here also means if you have `x = 2/3 + 4/6`, the substitution will go from `x` to `8/6` with the adding of fractions as substeps, which from a teaching perspective also feels weird to me.

My suggestion instead is:

1. take out the `const newScope = Substitute(options.scope);` above and keep the options exactly as they were passed in
2. here, instead of doing `const substeps = simplifyExpression(scope[symbolName]);` do `newNode = math.parse(scope[symbolName])`

I know it's a bit different from your idea, so we could chat more about this and make sure we're on the same page and both happy with whatever we decide to do. I think adding some tests that show multiple steps will also make it more clear what this changes. Let me know if you have more questions or comments about this!